### PR TITLE
Fixes after testing with GM

### PR DESCRIPTION
**Pull request by adizere**
Closes: #972 

## Description

______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer/src/supervisor.rs [hunk 1]
```diff
@@ -273,6 +273,11 @@ impl Supervisor {
             }
         }
 
+        // At least one chain runtime should be alive and kicking, otherwise bail.
```
Comment (ancazamfir) [inline relayer/src/supervisor.rs:276]: funny comment but what does it mean exactly?

Comment (adizere) [reply relayer/src/supervisor.rs:276]: I meant to say that if the vector of chain runtimes (`self.registry`) is empty, that means no chain runtime could be instantiated, so the supervisor (i) cannot spawn any worker and (ii) has no subscription to any chain to `select` on. I'll use more accurate language here!

Comment (ancazamfir) [reply relayer/src/supervisor.rs:276]: also https://github.com/informalsystems/ibc-rs/issues/972#issuecomment-847185935
I am not able to see that panic. What are the steps to see that? I tried to start hermes without any chains started and also tried to kill the chain procs while hermes is running.

**Review by ancazamfir — APPROVED** [review]
Ok, let's merge this so the client worker is fixed. Thanks again for finding this!!!

**Pull request merged**